### PR TITLE
HUM-641 Fixed areas where device weight < 0

### PR DESCRIPTION
**Pull request by creiht**

### CODE REVIEW

**Comment by dpgoetz** [comment]
dont know if its worth it but should we add a AllActiveDevices func to the ring to weed these out for you?

**Comment by creiht** [comment]
@dpgoetz That's a good question.  The other option is that we could add a util function (something like isActiveDevice) that would handle the checking.

**Comment by creiht** [comment]
ok I like @corystone's idea and will do that.

**Comment by gholt** [comment]
lgtm and thanks for adding my missing return :)

File: tools/ring.go [hunk 1]
```diff
@@ -453,7 +453,7 @@ func RingBuildCmd(flags *flag.FlagSet) {
 		totalWeight := float64(0)
 		devPartitions := make([]map[uint64]bool, len(devs))
 		for i, dev := range devs {
-			if dev != nil {
+			if dev != nil && dev.Weight >= 0 {
```
Comment (corystone) [inline tools/ring.go]: Shouldn't this be dev.Active()

File: tools/ring.go [hunk 2]
```diff
@@ -467,7 +467,7 @@ func RingBuildCmd(flags *flag.FlagSet) {
 		}
 
 		for i, dev := range devs {
-			if dev != nil {
+			if dev != nil && dev.Weight >= 0 {
```
Comment (corystone) [inline tools/ring.go]: And this?

File: tools/ring.go [hunk 3]
```diff
@@ -477,9 +477,9 @@ func RingBuildCmd(flags *flag.FlagSet) {
 
 		totalPairings := int64(0)
 		for i, dev1 := range devs {
-			if dev1 != nil {
+			if dev1 != nil && dev1.Weight >= 0 {
```
Comment (corystone) [inline tools/ring.go]: And this?

File: tools/ring.go [hunk 4]
```diff
@@ -477,9 +477,9 @@ func RingBuildCmd(flags *flag.FlagSet) {
...
 				for _, dev2 := range devs[i:] {
-					if dev2 != nil && dev1.Id != dev2.Id {
+					if dev2 != nil && dev2.Weight >= 0 && dev1.Id != dev2.Id {
```
Comment (corystone) [inline tools/ring.go]: Same

File: tools/ring.go [hunk 5]
```diff
@@ -490,9 +490,9 @@ func RingBuildCmd(flags *flag.FlagSet) {
 			totalSharesRequired = ring.PartitionCount()
 		}
 		for i, dev1 := range devs {
-			if dev1 != nil {
+			if dev1 != nil && dev1.Weight >= 0 {
 				for _, dev2 := range devs[i:] {
-					if dev2 != nil && dev1.Id != dev2.Id {
+					if dev2 != nil && dev2.Weight >= 0 && dev1.Id != dev2.Id {
```
Comment (corystone) [inline tools/ring.go]: Same

File: tools/ring.go [hunk 6]
```diff
@@ -552,7 +552,7 @@ func RingBuildCmd(flags *flag.FlagSet) {
 			}
 		}
 		for _, dev := range devs {
-			if dev != nil {
+			if dev != nil && dev.Weight >= 0 {
```
Comment (corystone) [inline tools/ring.go]: Same

File: tools/ring.go [hunk 7]
```diff
@@ -515,7 +515,7 @@ func RingBuildCmd(flags *flag.FlagSet) {
 		devices := make(map[string]bool)
 		primaryCounts := make(map[int][]int, len(devs))
 		for _, dev := range devs {
-			if dev != nil {
+			if dev != nil && dev.Weight >= 0 {
```
Comment (corystone) [inline tools/ring.go]: Same

**Pull request merged**